packages/core/src/components/canAccess/index.tsx

```diff
@@ -25,6 +25,19 @@ type CanAccessBaseProps = {
      * Content to show if access control returns `false`
      */
     fallback?: React.ReactNode;
+    /**
+     * Callback function to be called if access control returns `false`
+     */
+    onUnauthorized?: (props: {
+        resource?: string;
+        reason?: string;
+        action: string;
+        params: {
+            resource?: IResourceItem & { children?: ITreeMenu[] };
+            id?: BaseKey;
+            [key: string]: any;
+        };
+    }) => void;
```

Comment:
```suggestion
    }) => Promise<void> | void;
```

I think we should allow it to be an async function, even though we won't await it.